File: robocode-tankroyale-bot-api-csharp/src/SpinBot.cs
using Robocode.TankRoyale.BotApi;

namespace Robocode.TankRoyale.Sample.Bots
{
  /// <summary>
  /// SpinBot - a sample bot.
  ///
  /// Moves in a circle, firing hard when an enemy is detected.
  /// </summary>
  public class SpinBot : Bot
  {
    static void Main2(string[] args)
    {
      new SpinBot().Start();
    }

    SpinBot() : base(BotInfo.FromJsonFile("bot-settings.json")) { }

    // This method runs our bot program, where each command is executed one at a time
    public override void Run()
    {
      SetBodyColor("#00F"); // blue
      SetTurretColor("#00F"); // blue
      SetRadarColor("#000"); // black
      SetScanColor("#FF0"); // yellow

      // Repeat while bot is running
      while (IsRunning)
      {
        // Tell the game that when we take move, we'll also want to turn right... a lot.
        SetTurnRight(10000);
        // Limit our speed to 5
        SetMaxSpeed(5);
        // Start moving (and turning)
        Forward(10000);
      }
    }

    // OnScannedBot: Fire hard when scanning another bot!
    public override void OnScannedBot(ScannedBotEvent evt)
    {
      Fire(3); // Fire the cannon!
    }

    // OnHitBot: If it's our fault, we'll stop turning and moving, so we need to turn again to keep spinning.
    public override void OnHitBot(BotHitBotEvent e)
    {
      double direction = CalcDirection(e.X, e.Y);
      double bearing = CalcBearing(direction);
      if (bearing > -10 && bearing < 10)
      {
        Fire(3);
      }
      if (e.IsRammed)
      {
        TurnRight(10);
      }
    }
  }
}

File: sample-bots/java/SpinBot/src/main/java/dev/robocode/tankroyale/sample/bots/SpinBot.java
package dev.robocode.tankroyale.sample.bots;

import dev.robocode.tankroyale.botapi.Bot;
import dev.robocode.tankroyale.botapi.BotInfo;
import dev.robocode.tankroyale.botapi.events.HitBotEvent;
import dev.robocode.tankroyale.botapi.events.ScannedBotEvent;

import java.io.IOException;

/**
 * SpinBot - a sample bot, original version by Mathew Nelson for Robocode.
 *
 * <p>Moves in a circle, firing hard when an enemy is detected.
 */
public class SpinBot extends Bot {

  public static void main(String[] args) throws IOException {
    new SpinBot().start();
  }

  protected SpinBot() throws IOException {
    super(BotInfo.fromFile("/SpinBot.json"));
  }

  /** SpinBot's run method - Move in a circle */
  @Override
  public void run() {
    setBodyColor("#00F"); // blue
    setTurretColor("#00F"); // blue
    setRadarColor("#000"); // black
    setScanColor("#FF0"); // yellow

    // Repeat while bot is running
    while (isRunning()) {
      // Tell the game that when we take move, we'll also want to turn right... a lot.
      setTurnLeft(10_000);
      // Limit our speed to 5
      setMaxSpeed(5);
      // Start moving (and turning)
      forward(10_000);
    }
  }

  /** onScannedBot: Fire hard when scanning another bot! */
  @Override
  public void onScannedBot(ScannedBotEvent e) {
    fire(3); // Fire the cannon!
  }

  /**
   * onHitBot: If it's our fault, we'll stop turning and moving, so we need to turn again to keep spinning.
   */
  @Override
  public void onHitBot(HitBotEvent e) {
    double direction = directionTo(e.getX(), e.getY());
    double bearing = calcBearing(direction);
    if (bearing > -10 && bearing < 10) {
      fire(3);
    }
    if (e.isRammed()) {
      turnLeft(10);
    }
  }
}